<launch>
  <!-- configure console output verbosity mode:debug_console.conf or std_console.conf -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find spinnaker_sdk_camera_driver)/cfg/std_console.conf"/>

  <!-- acquisition.lauynch -->
  <arg name="binning"		        default="1"	doc="Binning for cameras, when changing from 2 to 1 cameras need to be unplugged and replugged"/>
  <arg name="color"		          default="true"	doc="Should color images be used (only works on models that support color images)"/>
  <arg name="exp"               default="-1"	doc="Exposure setting for cameras"/>
  <arg name="gain"              default="-1" doc="Gain Setting for cameras"/>
  <arg name="gamma"             default="-1" doc="Gamma Correction for cameras"/>
  <arg name="balance"           default="-1" doc="White Balance Setting for cameras for color only"/>
  <arg name="output"            default="screen"	doc="display output to screen or log file"/>
  <arg name="save"              default="false"	doc="flag whether images should be saved or not"/>
  <arg name="save_path"         default="~"	doc="location to save the image data"/>
  <arg name="save_type"         default="tiff"	doc="Type of file type to save to when saving images locally: tiff, bmp, jpeg etc." />
  <arg name="to_ros"            default="true"	doc="Flag whether images should be published to ROS" />
  <arg name="config_file"       default="$(find spinnaker_sdk_camera_driver)/params/L_params.yaml" doc="File specifying the parameters of the camera_array"/>

  <!-- load the acquisition node -->
  <node pkg="spinnaker_sdk_camera_driver" type="acquisition_node" name="acquisition_node" output="screen"
	args="">

    <!-- load the acquisition node parameters file.  Note any parameters provided in this file will 
	 override what is in the yaml file.  Thus use it to set parameters camer_array configuration params -->
    <rosparam command="load"        file="$(arg config_file)" />

    <!-- Load parameters onto server using argument or default values above -->
    <param name="exp"               value="$(arg exp)" />
    <param name="binning"           value="$(arg binning)" />
    <param name="color"             value="$(arg color)" />
    <param name="gain"              value="$(arg gain)"/>
    <param name="balance"              value="$(arg balance)"/>
    <param name="gamma"              value="$(arg gamma)"/>
    <param name="save"              value="$(arg save)" />
    <param name="save_type"         value="$(arg save_type)" />
    <param name="save_path"         value="$(arg save_path)" />
    <param name="to_ros"            value="$(arg to_ros)"/>

  </node>    
 
</launch>
